src/main/scala/uk/gov/hmrc/domain/Random.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2015 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.domain
+
+class Random(random: scala.util.Random = new scala.util.Random) {
```

Comment:
I think this class being called the same as the class of one of it's constructor arguments is a bit confusing and will also make it awkward to import for a client?

Perhaps we could call it `Generator`?

I'm thinking:

```scala
trait Generate[T] {
  def random: scala.util.Random
  def generateNext: T
}

object Nino extends Generate[Nino] {
...
}
```

There might be a case for having `Generator[Nino]` available as a separate trait as well.